server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -93,7 +98,18 @@
      */
     public SeLionRemoteProxy(RegistrationRequest request, Registry registry) throws IOException {
         super(request, registry);
-
+        String urlString = request.getConfiguration().get("url").toString();
+        URL url = new URL(urlString);
+        for (DesiredCapabilities desiredCapabilities : request.getCapabilities()) {
+            Map<String, ?> capabilitiesMap = desiredCapabilities.asMap();
+            String browserName = capabilitiesMap.get(CapabilityType.BROWSER_NAME).toString();
+            String maxInstancesAsString = capabilitiesMap.get("maxInstances").toString();
+            if (StringUtils.isNotBlank(browserName) && StringUtils.isNotBlank(maxInstancesAsString)) {
+                int maxInstances = Integer.valueOf(maxInstancesAsString);
+                BrowserInformationCache cache = BrowserInformationCache.getInstance();
+                cache.updateBrowerInfo(url, browserName, maxInstances);
+            }
+        }
```

Comment:
There's no sense in doing any of this if the hub does not have the `GridStatistics` servlet configured.. Can we please make this code trigger based on the hubs configuration? 

Also, will you please extra method refactor and comment this code?